worlds/yoshisisland/Options.py

```diff
@@ -0,0 +1,271 @@
+       You can check this in-game by pressing SELECT while in any level."""
+    display_name = "Bosses Required for 6-8 Clear"
+    range_start = 0
+    range_end = 11
+    default = 0
+
+class BowserDoor(Choice):
+    """This will set which route you take through 6-8.
+    Manual: You go through the door that you hit with an egg, as normal.
+    Doors: Route will be forced to be the door chosen here, regardless of which door you hit.
+    Gauntlet: You will be forced to go through all 4 routes in order before the final hallway."""
+    display_name = "Bowser's Castle Doors"
+    option_manual = 0
+    option_door_1 = 1
+    option_door_2 = 2
+    option_door_3 = 3
+    option_door_4 = 4
+    option_gauntlet = 5
+    default = 0
+
+class BossShuffle(Toggle):
+    """This whill shuffle which boss each boss door will lead to. Each boss can only appear once, and Baby Bowser is left alone."""
+    display_name = "Boss Shuffle"
+
+class LevelShuffle(Choice):
+    """Disabled: All levels will appear in their normal location.
+    Bosses Guranteed: All worlds will have a boss on -4 and -8.
+    Full: Worlds may have more than 2 or no bosses in them.
+    Regardless of the setting, 6-8 and Extra stages are not shuffled."""
+    display_name = "Level Shuffle"
+    option_disabled = 0
+    option_bosses_guranteed = 1
+    option_full = 2
+    default = 0
+
+class YoshiColors(Choice):
+    """Sets the Yoshi color for each level.
+    Normal will use the vanilla colors.
+    Random order will generate a random order of colors that will be used in each level. The stage 1 color will be used for Extra stages, and 6-8.
+    Random color will generate a random color for each stage.
+    Singularity will use a single color defined under 'Singularity Yoshi Color' for use in all stages."""
+    display_name = "Yoshi Colors"
+    option_normal = 0
+    option_random_order = 1
+    option_random_color = 2
+    option_singularity = 3
+    default = 0
+
+class SinguColor(Choice):
+    """Sets which color Yoshi will be if Yoshi Colors is set to singularity."""
+    display_name = "Singularity Yoshi Color"
+    option_green = 0
+    option_pink = 1
+    option_cyan = 3
+    option_yellow = 2
+    option_purple = 4
+    option_brown = 5
+    option_red = 6
+    option_blue = 7
+    default = 0
+
+class BabySound(Choice):
+    """Change the sound that Baby Mario makes when not on Yoshi."""
+    display_name = "Mario Sound Effect"
+    option_normal = 0
+    option_disabled = 1
+    option_random_sound_effect = 2
+    default = 0
+
+class TrapsEnabled(Toggle):
+    """Will place traps into the item pool. 
+       Traps have a variety of negative effects, and will only replace filler items."""
+    display_name = "Traps Enabled"
+
+class TrapPercent(Range):
+    """Percentage of the item pool that becomes replaced with traps."""
+    display_name = "Trap Chance"
+    range_start = 0
+    range_end = 100
+    default = 10
+
+#class EnableScrets(Range):

```

Comment:
I know this is commented out, but assuming the intent is to enable these eventually, I think this one isn't named correctly